README.md

```diff
@@ -15,251 +18,329 @@ The role can be used to configure:
 - Infiniband interfaces
 - IP configuration
 
-General
--------
-
-The role supports two providers: `nm` and `initscripts`. The provider can be
+Introduction
+------------
+The  `network` role supports two providers: `nm` and `initscripts`. `nm` is
+used by default in RHEL7 and `initscripts` in RHEL6. These providers can be
 configured per host via the [`network_provider`](#provider) variable. In
 absence of explicit configuration, it is autodetected based on the
-distribution. The `nm` provider is used by default on RHEL7 and `initscripts`
-on RHEL6. However, note that the provider is not tied to a certain
-distribution, given that the required API is available. For `nm` this means
-that at least version 1.2 of NetworkManager's API is available. For
-`initscripts`, it requires the legacy network service as commonly available on
-Fedora/RHEL.
-
-For each host a list of networking profiles can be configured via the
-`network_connections` variable.
+distribution. However, note that either `nm` or `initscripts` is not tied to a certain
+distribution. The `network` role works everywhere the required API is available.
+This means that `nm` requires at least NetworkManager's API version 1.2 available.
+For `initscripts`, the legacy network service is required as used in Fedora or RHEL.
 
```

Comment:
This does not make sense in the new version, because we start talking about profiles below without introducing them (after the paragraph above was deleted or moved to a different place).